src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -1121,18 +1121,16 @@ private void ImportDataMembers()
                             DataMember memberContract = new DataMember(member);
 
                             memberContract.Name = DataContract.EncodeLocalName(member.Name);
-                            object[] optionalFields = null;
-                            if (optionalFields == null || optionalFields.Length == 0)
```

Comment:
Opened https://github.com/dotnet/corefx/issues/11477 for tracking adding the implementation back.